executor/adapter.go

```diff
@@ -288,6 +289,18 @@ func (a *ExecStmt) RebuildPlan(ctx context.Context) (int64, error) {
 	return a.InfoSchema.SchemaMetaVersion(), nil
 }
 
+func (a *ExecStmt) wrapAndSetPlanLabelForTopSQL(ctx context.Context) context.Context {
+	if a.Plan == nil || !variable.TopSQLEnabled() {
+		return ctx
+	}
+	_, sqlDigest := a.Ctx.GetSessionVars().StmtCtx.SQLDigest()
+	normalizedPlan, planDigest := getPlanDigest(a.Ctx, a.Plan)
+	if len(planDigest) > 0 {
+		ctx = topsql.SetSQLAndPlanLabels(ctx, sqlDigest.String(), planDigest, normalizedPlan)
```

Comment:
Seems that we can accept bytes in `SetSQLAndPlanLabels`? Then the string size is reduced by 50%.